portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,42 +34,45 @@
 @WebServlet("/comments")
 public class DataServlet extends HttpServlet {
 
-  private ArrayList<String> comments;
-
-  @Override
-  public void init() {
-    comments = new ArrayList<String>();
-    comments.add("Wow, Obi nice website!");
-    comments.add("I love the color scheme");
-    comments.add("We love you Obi!");
-  }
-
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String json = convertToJson(comments);
-    
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    ArrayList<Comment> commentList = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String text = (String) entity.getProperty("text");
+      long timestamp = (long) entity.getProperty("timestamp");
+
+      Comment comment = new Comment(id, text, timestamp);
+      commentList.add(comment);
+    }
+
+    Gson gson = new Gson();
+
     response.setContentType("application/json;");
-    response.getWriter().println(json);
+    response.getWriter().println(gson.toJson(commentList));
   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Get the input from the form.
-    String comment = request.getParameter("comment-input");
-    if (!comment.isBlank()) { 
-      comments.add(comment); 
+    String text = request.getParameter("comment-input");
+    if (!text.isBlank()) { 
+      long timestamp = System.currentTimeMillis();
```

Comment:
Call this something like `currentTime` or `commentPostTime` rather than `timestamp`. If variables have more descriptive names readers can usually tell what they do without looking back to their original declaration. This makes code much easier to read.

In this code it doesn't hinder readability too much if you call this variable something generic like `timestamp` because there aren't that many variables. However, you should rename it anyway so you get in the habit of being very descriptive with variable names. When there are 10s of variables it's a lot easier to remember what each one does if they have very specific and descriptive names.

Reply:
```suggestion
      long commentPostTime = System.currentTimeMillis();
```